File: src/Training.FileExplorer.ClientApp/src/infrastructure/models/entities/StorageFile.ts
import { IStorageEntry } from "@/infrastructure/models/entities/IStorageEntry";
import type { StorageEntryType } from "@/infrastructure/models/entities/StorageItemType";

export class StorageFile implements IStorageEntry {
     name!: string;
     path!: string;
     directoryPath!: string;
     size!: number;
     extension!: string;
     entryType!: StorageEntryType;
}

File: src/FromScratch.ServerApp/FileExplorer.Application/FileStorage/Models/StorageFile.cs
namespace FileExplorer.Application.FileStorage.Models;

public class StorageFile : IStorageEntry
{
    public string Name { get; set; } = string.Empty;

    public string Path { get; set; } = string.Empty;

    public string? DirectoryPath { get; set; } = string.Empty;

    public long Size { get; set; }

    public string Extension { get; set; } = string.Empty;

    public StorageEntryType EntryType { get; set; } = StorageEntryType.File;
}